src/mapping/FGreedyCutSolver.hpp

```diff
@@ -0,0 +1,243 @@
+std::pair<int, double> FGreedySolver<RADIAL_BASIS_FUNCTION_T>::select(const Eigen::VectorXd &residual) const
+{
+  Eigen::Index maxIndex;
+  double maxValue = residual.maxCoeff(&maxIndex);
+  return {maxIndex, maxValue};
+}
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+Eigen::MatrixXd FGreedySolver<RADIAL_BASIS_FUNCTION_T>::buildEvaluationMatrix(const std::vector<int> &greedyIDs) const
+{
+  Eigen::MatrixXd matrixA(greedyIDs.size(), _outputVertices.size());
+
+  for (size_t i = 0; i < greedyIDs.size(); i++) 
+  {
+    const auto &u = _inputVertices.at(greedyIDs.at(i)).rawCoords();
+    for (size_t j = 0; j < _outputVertices.size(); j++) 
+    {
+      const auto  &v = _outputVertices.at(j).rawCoords();
+      const double squaredDifference = computeSquaredDifference(u, v, _activeAxis);
+      matrixA(i, j) = _basisFunction.evaluate(std::sqrt(squaredDifference));
+    }
+  }
+
+  return matrixA;
+}
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+Eigen::VectorXd FGreedySolver<RADIAL_BASIS_FUNCTION_T>::recalculateResidual(const std::vector<int> &greedyIDs, const Eigen::MatrixXd &cut, Eigen::MatrixXd &kernelMatrix, const Eigen::VectorXd &inputData, Eigen::VectorXd &c) const {
+  int n = greedyIDs.size();
+  const auto &v = _inputVertices.at(greedyIDs.at(n-1)).rawCoords();
+  for (size_t i = 0; i < _inSize; i++) 
+  {
+    const auto &u = _inputVertices.at(i).rawCoords();
+    const double squaredDifference = computeSquaredDifference(u, v, _activeAxis);
+    kernelMatrix(i, n-1) = _basisFunction.evaluate(std::sqrt(squaredDifference));
+  }
+  const double a = (cut.block(n-1,0,1,n) * inputData(greedyIDs))(0,0);
+  c.head(n) += a * cut.block(n-1,0,1,n).transpose();
+  return (inputData - kernelMatrix.block(0,0,_inSize, n) * c.head(n)).cwiseAbs();
+}
+
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void FGreedySolver<RADIAL_BASIS_FUNCTION_T>::updateKernelVector(const std::vector<int> greedyIDs, Eigen::VectorXd &kernelVector, const mesh::Vertex &x) const
+{
+  for (size_t j = 0; j < greedyIDs.size(); j++)
+  {
+    const auto &y = _inputVertices.at(greedyIDs.at(j)).rawCoords();
+    kernelVector(j) = _basisFunction.evaluate(std::sqrt(computeSquaredDifference(x.rawCoords(), y, _activeAxis))); 
+  }
+}
+
+
+// ---- Initialization ---- //
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+template <typename IndexContainer>
+FGreedySolver<RADIAL_BASIS_FUNCTION_T>::FGreedySolver(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, const IndexContainer &inputIDs,
+                                                      const mesh::Mesh &outputMesh, const IndexContainer &outputIDs, std::vector<bool> deadAxis, Polynomial polynomial)
+  : _inSize(inputMesh.vertices().size()), 
+    _outSize(outputMesh.vertices().size()), 
+    _inputVertices(inputMesh.vertices()),
+    _outputVertices(outputMesh.vertices()),
+    _basisFunction(basisFunction)
+{
+  precice::profiling::Event initEvent("FGreedy.initialize", profiling::Synchronize);
```

Comment:
```suggestion
  precice::profiling::Event initEvent("FGreedy.initialize");
```
Note that the current mapping class (`RadialBasisFctMapping`), which is instantiated with this solver uses a gather-scatter approach to collect distributed data and the solver is only executed in serial. The `profiling::Synchronize` is an [MPI barrier](https://docs.open-mpi.org/en/v5.0.x/man-openmpi/man3/MPI_Barrier.3.html). Thus, the sync would lead to a deadlock here in case of parallel execution with synchronization.